### PR TITLE
Add comet functionality to the keras model

### PR DESCRIPTION
**Pull request by nthain**
* Add code to track experiments in comet.ml
* Requires that you load an API key from a local file 'comet_api_key.txt'
* Defines list of hyperparameters to track, though tracked metrics are inferred from the graph

### CODE REVIEW

**Comment by ericagreene** [comment]
This closes https://github.com/conversationai/conversationai-models/issues/56 😃

File: kaggle-classification/bin/run_keras_local.sh [hunk 1]
```diff
@@ -20,4 +22,5 @@ python -m keras_trainer.model \
 	--validation_path=${INPUT_PATH}/validation.csv \
 	--embeddings_path=${INPUT_PATH}/glove.6B/glove.6B.100d.txt \
 	--job-dir=${OUTPUT_PATH} \
-	--log_path=${LOG_PATH}
+	--log_path=${LOG_PATH} \
+    --comet_key ${COMET_KEY}
```
Comment (jjtan) [inline kaggle-classification/bin/run_keras_local.sh]: format

File: kaggle-classification/keras_trainer/model.py [hunk 2]
```diff
@@ -113,6 +115,7 @@ def train(self, train):
 
   def predict(self, texts):
     data = self._prep_texts(texts)
+    self._load_model()
```
Comment (jjtan) [inline kaggle-classification/keras_trainer/model.py]: Why is this necessary? The model i loaded when the class is instantiated and then loaded whenever the model is trained.

File: kaggle-classification/keras_trainer/model.py [hunk 3]
```diff
@@ -182,13 +187,25 @@ def _setup_embeddings_matrix(self):
...
   FLAGS = parser.parse_args()
 
   hparams = DEFAULT_HPARAMS
   hparams.learning_rate = FLAGS.learning_rate
   hparams.dropout_rate = FLAGS.dropout_rate
   hparams.batch_size = FLAGS.batch_size
 
+  if FLAGS.comet_key:
+    experiment = Experiment(api_key=FLAGS.comet_key, 
+      project_name='comet_trial_run',
```
Comment (jjtan) [inline kaggle-classification/keras_trainer/model.py]: should this be flag? up to you

File: kaggle-classification/keras_trainer/model.py [hunk 4]
```diff
@@ -182,13 +187,25 @@ def _setup_embeddings_matrix(self):
...
+  if FLAGS.comet_key:
+    experiment = Experiment(api_key=FLAGS.comet_key, 
+      project_name='comet_trial_run', 
+      auto_param_logging=False,
+      parse_args=False)
+    experiment.log_multiple_params(hparams.values())
+    experiment.log_parameter('test_data_path', FLAGS.train_path)
+    experiment.log_parameter('valid_data_path', FLAGS.validation_path)
+    experiment.log_parameter('embeddings_path', FLAGS.embeddings_path)
+    experiment.log_parameter('model_path', FLAGS.job_dir)
```
Comment (jjtan) [inline kaggle-classification/keras_trainer/model.py]: add model

File: kaggle-classification/keras_trainer/model.py (hunk 2 above)
Comment (nthain) [reply kaggle-classification/keras_trainer/model.py]: Good point. Removed.

File: kaggle-classification/keras_trainer/model.py (hunk 3 above)
Comment (nthain) [reply kaggle-classification/keras_trainer/model.py]: Yes, we want this to work even if you don't have a comet API key.

File: kaggle-classification/keras_trainer/model.py (hunk 4 above)
Comment (nthain) [reply kaggle-classification/keras_trainer/model.py]: Done.

File: kaggle-classification/keras_trainer/model.py (hunk 3 above)
Comment (jjtan) [reply kaggle-classification/keras_trainer/model.py]: I was referring to project_name. But actually it sounds like we might all be sharing one project?
